test/client/Query.test.tsx

```diff
@@ -1174,7 +1200,161 @@ describe('Query component', () => {
+      query people {
+        allPeople(first: 1) {
+          people {
+            name
+          }
+        }
+      }
+    `;
+
+    const link = new ApolloLink((o, f) => {
+      done.fail(new Error('query ran even though skip present'));
+      return f ? f(o) : null;
+    }).concat(mockSingleLink());
+    const client = new ApolloClient({
+      link,
+      cache: new Cache({ addTypename: false }),
+    });
+
+    wrapper = mount(
+      <ApolloProvider client={client}>
+        <Query query={query} skip={() => true}>
+          {result => {
+            expect(result).toEqual(undefined);
+            done();
+
+            return null;
+          }}
+        </Query>
+      </ApolloProvider>,
+    );
+  });
+
+  it('continues to not subscribe to a skipped query when props change', done => {
+    const query: DocumentNode = gql`
+      query people {
+        allPeople(first: 1) {
+          people {
+            name
+          }
+        }
+      }
+    `;
+
+    const link = new ApolloLink((o, f) => {
+      done.fail(new Error('query ran even though skip present'));
+      return f ? f(o) : null;
+    }).concat(mockSingleLink());
+
+    const client = new ApolloClient({
+      link,
+      cache: new Cache({ addTypename: false }),
+    });
+
+    class Container extends React.Component<
+      {},
+      { variables: { number: number } }
+    > {
+      state = {
+        variables: {
+          number: 123,
+        },
+      };
+
+      componentDidMount() {
+        this.setState({ variables: { number: 999 } });
+      }
+
+      render() {
+        return (
+          <Query
+            query={query}
+            skip={() => true}
+            variables={this.state.variables}
+          >
+            {result => {
+              expect(result).toBeUndefined();
+
+              if (this.state.variables.number === 999) {
+                done();
+              }
+
+              return null;
+            }}
+          </Query>
+        );
+      }
+    }
+
+    wrapper = mount(
+      <ApolloProvider client={client}>
+        <Container />
+      </ApolloProvider>,
+    );
+  });
+
+  fit('allows you to skip then unskip a query with top-level syntax', done => {
+    const query: DocumentNode = gql`
+      query people {
+        allPeople(first: 1) {
+          people {
+            name
+          }
+        }
+      }
+    `;
+    const data = { allPeople: { people: [{ name: 'Luke Skywalker' }] } };
+    const link = mockSingleLink({
+      request: { query },
+      result: { data },
+    });
+
+    const client = new ApolloClient({
+      link,
+      cache: new Cache({ addTypename: false }),
+    });
+
+    class Container extends React.Component<{}, { skip: boolean }> {
+      state = {
+        skip: true,
+      };
+
+      componentDidMount() {
+        this.setState({ skip: false });
+      }
+
+      render() {
+        return (
+          <Query query={query} skip={() => this.state.skip}>
```

Comment:
I don't think skip should be a function, since you have props in scope, it should just be a boolean

Reply:
This makes sense to me! Will update